src/index.ts

```diff
@@ -284,6 +284,17 @@ const plugin: JupyterFrontEndPlugin<void> = {
           return;
         }
 
+        const currentKernel = panel.sessionContext.session?.kernel?.name || '';
+        const currentKernelDisplay = KERNEL_DISPLAY_NAMES[currentKernel] || currentKernel;
+        const targetKernelDisplay = KERNEL_DISPLAY_NAMES[kernel] || kernel;
```

Comment:
I guess these can be moved into the `if` block, since we only need them if we enter that block.